pkg/controller/controller.go

```diff
@@ -456,19 +461,12 @@ func (c *M3DBController) handleClusterUpdate(cluster *myspec.M3DBCluster) error
 				return err
 			}
 
-			_, err = c.kubeClient.AppsV1().StatefulSets(cluster.Namespace).Create(sts)
+			set, err := c.kubeClient.AppsV1().StatefulSets(cluster.Namespace).Create(sts)
 			if err != nil {
-				if kerrors.IsAlreadyExists(err) {
```

Comment:
We shouldn't be able to run into this here anymore because we're now checking that the StatefulSet exists in `getChildStatefulSets`.